CRM/Report/Form/Contact/Detail.php

```diff
@@ -975,4 +976,44 @@ public function alterComponentDisplay(&$componentRows) {
     }
   }
 
+  /**
+   * Function to alter from clause by adding checks for permissoned financial type.
+   *
+   * @param string $component
+   *
+   * @return string
+   */
+  public function addPermissionedFromClause($component) {
+    if (is_null($this->financialTypes)) {
+      return NULL;
+    }
+    $financialTypes = 0;
+    if (!empty($this->financialTypes)) {
+      $financialTypes = implode(',', array_keys($this->financialTypes));
+    }
+    $clause = "financial_type_id IN ({$financialTypes})";
+    $whereClause = NULL;
+
+    switch ($component) {
```

Comment:
@eileenmcnaughton @JoeMurray I tried refactoring this but it seems like a big change to make. The components are already being iterated through a foreach https://github.com/civicrm/civicrm-core/pull/8891/files#diff-3061684a1d8e2be5d9854b53781f7e4cR610 
We just added a whereclause for different components and added a join to restrict only those financial types. Do we still need to rewrite this code?